### PR TITLE
Add cron service and cleanup script

### PR DESCRIPTION
**Pull request by nrobinaubertin**
### Overview

Synapse can leave empty directories in the media directory after deleting old/unused files. There is currently no option to remove them and they can accumulate, using inodes and space.

This PR adds a cron service inside the workload container to regularly launch a script meant to delete those empty directories.  
The PR was structure to simplify the future addition of scripts that should be handled by the cron service. Just add them in the corresponding `cron.{hourly,daily,weekly,monthly}` directory.

Remarks:
- I tried the cleanup script on a VPS with half a million empty directories: CPU and memory usage were not noticeable and I tweaked the script to lower the IOPS (see the comments inside it for explanations)

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Manging Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: synapse_rock/cron/cron.weekly/remote_content_empty_directory_cleanup.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import json
+
+# We assume that pyyaml is present thanks to synapse
+import yaml
+
+# This function is meant to fail if the media_store_path can't be found
+def load_media_store_path(file_path: str) -> str:
+    with open(file_path, 'r', encoding='utf-8') as file:
+        data = yaml.safe_load(file)
+        return data['media_store_path']
```
[the remark this answers is not in the document]
Comment (nrobinaubertin) [reply synapse_rock/cron/cron.weekly/remote_content_empty_directory_cleanup.py]: Is this true while removing empty directories ? I feel like the records in the database would be for files

**Review by javierdelapuente — DISMISSED** [review]
LGTM

File: synapse_rock/cron/cron.weekly/remote_content_empty_directory_cleanup.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+
+# This function is meant to fail if the media_store_path can't be found
+def load_media_store_path(file_path: str) -> str:
+    with open(file_path, "r", encoding="utf-8") as file:
+        data = yaml.safe_load(file)
+        return data["media_store_path"]
+
+
+def delete_empty_dirs(path: str) -> None:
```
Comment (merkata) [inline synapse_rock/cron/cron.weekly/remote_content_empty_directory_cleanup.py:38]: why no docstrings?

**Review by merkata — APPROVED** [review]
:100:

File: synapse_rock/cron/cron.weekly/remote_content_empty_directory_cleanup.py (hunk 2 above)
Comment (merkata) [reply synapse_rock/cron/cron.weekly/remote_content_empty_directory_cleanup.py:38]: can be done in another PR though, code is fine imo

**Pull request merged**